earth_enterprise/src/common/khCache.h

```diff
@@ -217,6 +218,7 @@ class khCache {
     limitCacheMemory = enabled;
     maxCacheMemory = maxMemory;
   }
+  void setPruneConditionVal(bool val) { pruneConditionMet = val; }
```

Comment:
You shouldn't need a new function and variable to implement this. The `Add` and `Remove` function already have an option that lets you decide whether to prune.